app/assets/js/rtm-upload-terms.js

```diff
@@ -6,43 +7,93 @@
  * By: Malav Vasita <malav.vasita@rtcamp.com>
  */
 if ( 'object' === typeof rtMediaHook ) {
+
     /**
-     * Check for the terms checkbox to be checked before the media is uploaded.
-     * Show warning if unchecked else proceed.
+     * Check terms condition checkbox before uploading files.
      */
-    rtMediaHook.register( 'rtmedia_js_upload_file', function ( args ) {
+    rtMediaHook.register( 'rtmedia_js_before_upload', function ( args ) {
 
-        if ( false === args ) {
+        if ( false === args || 'undefined' === typeof args.uploader.settings.multipart_params || 'undefined' === typeof args.src ) {
 
             return args;
         }
 
-        /**
-         * Show the warning message if the terms checkbox is enabled in the admin panel.
-         * Extra check if the HTML not exists, or has been removed intentionally.
-         *
-         * @author Adarsh Verma <adarsh.verma@rtcamp.com>
-         */
-        var activity_terms_enabled = rtmedia_upload_terms_data.activity_terms_enabled;
+        var multipart_params = args.uploader.settings.multipart_params;
+        var request_key = false;
+        var terms_key = false;
+        var isTermsEnabled = false;
 
-        if ( 'false' === activity_terms_enabled ) {
+        if ( 'activity' === args.src ) {
+            request_key = 'activity_terms_condition_request';
+            terms_key = 'activity_terms_condition';
 
-            return false;
+            if ( 'true' === rtmedia_upload_terms_data.activity_terms_enabled ) {
```

Comment:
`rtmedia_upload_terms_data.activity_terms_enabled` this return `string/boolean` ?

Reply:
It's localized so it'll be a string